src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -1350,9 +1377,16 @@ module T = struct
         ~f:(fun (seq, count) w ->
           match get_completed_work w with
           | Some cw_checked ->
-              Continue
-                ( Sequence.append seq (Sequence.singleton cw_checked)
-                , One_or_two.length cw_checked.proofs + count )
+              (*If new provers can't pay the account-creation-fee then discard their work. To encourage using an existing account for snark workers*)
+              if
+                (not (is_new_account cw_checked.prover))
+                || Currency.Fee.(
+                     cw_checked.fee >= Coda_compile_config.account_creation_fee)
+              then
+                Continue
+                  ( Sequence.append seq (Sequence.singleton cw_checked)
+                  , One_or_two.length cw_checked.proofs + count )
+              else Stop (seq, count)
```

Comment:
Wouldn't you want to Continue in this case, just without updating `seq` and `count`? IIUC, this code will cease including any completed work as soon as it encounters a single work which cannot afford the account creation fee.

Reply:
You need a continuous sequence of work. That's what the `Stop` in the `None` case below is also for (when you don't get completed work for a specific statement)